dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+'''
+'''
+
+from __future__ import absolute_import
+
+# standard imports
+import logging
+import uuid
+import time
+import os
+import json
+from datetime import datetime
+
+# internal imports
+from dripline import core
+from dragonfly.implementations import EthernetProvider
+
+
+__all__ = []
+
+logger = logging.getLogger(__name__)
+
+__all__.append('PsyllidProvider')
+class PsyllidProvider(core.Provider, core.Spime):
+    '''
+    Provider for direct communication with Psyllid 
+    '''
+    def __init__(self,
+                 psyllid_queue='psyllid',
+                 set_condition_list = [],
+                 **kwargs):
+
+	core.Provider.__init__(self, **kwargs)        
+        self.psyllid_queue = psyllid_queue
+        self._set_condition_list = set_condition_list
+        self.status = None
+        self.status_value = None
+               
+        self.channel_dict = {'a': 0, 'b': 1, 'c': 2}
+        self.freq_dict = {'a': None, 'b': None, 'c': None}
+        self.mode = None
+        
+        
+    def _finish_configure(self):
+        try:
+            if self.request_status()==0:
+                self.activate()
+        except:
+            raise core.exceptions.DriplineGenericDAQError('Psyllid instance with this queue is not responding')
+
+        self.mode = 'streaming'
+        try:
+            self.set_default_central_frequencies()
+        except:
+            self.mode = 'triggered'
+            self.set_default_central_frequencies()
+        self.get_number_of_streams()
+        
+        
+    def set_default_central_frequencies(self):
+        freq_dict = {'a': 800e6, 'b': 800e6, 'c': 800e6}
+        self.set_all_central_frequencies(freq_dict)
+        
+
+    def request_status(self):
+        logger.info('Checking Psyllid status')
+        
+        try:
```

Comment:
same, a try/except with a blanket except is pretty dangerous. What is producing the the exception you would catch and what exception(s) do you expect? We should be able to check the return code on the get().